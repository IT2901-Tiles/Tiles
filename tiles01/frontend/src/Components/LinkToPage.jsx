import React, { useState } from 'react';
import { Link } from 'react-router-dom';

// Component for link on images in ChosenCardsPage.jsx
function LinkToPage(props) {
    const [path] = useState(props.target)
    const [category] = useState(props.category)
    const [nameOfCard] = useState(props.nameOfCard) // name of card, e.g. trigger, thing, feedback
    const [classNameCard] = useState(props.classNameCard) // classname of the card

    let colour = "#F08A00";

    if (category === "things") {
        colour = "#D64539"
    } else if (category === "feedback") {
        colour = "#FFCD00"
    } else if (category === "animation") {
        colour = "#84AD64"
    }

    return (
        <Link
            to={{
                pathname: path,
                state: { category: category, color: colour, card: props.card }
            }}>
            <img alt={nameOfCard} src={nameOfCard} className={classNameCard} />
        </Link>
    );
}

export default LinkToPage;
