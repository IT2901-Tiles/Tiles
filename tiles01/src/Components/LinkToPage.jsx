import React, { useState } from 'react';
import { Link } from 'react-router-dom';

// Component for link on images in ChosenCardsPage.jsx

function LinkToPage(props) {
    const [path] = useState(props.target)
    const [category] = useState(props.category)
    const [nameOfCardType] = useState(props.nameOfCardType)


    return (
        <Link
            to={{
                pathname: path,
                state: { category: category, card:props.card}
            }}>
            <img src={nameOfCardType} className="chosenTrigger" />
        </Link>
    );
}

export default LinkToPage
