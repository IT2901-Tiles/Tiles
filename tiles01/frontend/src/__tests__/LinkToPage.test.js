import {configure, shallow} from "enzyme"
import Adapter from 'enzyme-adapter-react-16';
import {cleanup} from '@testing-library/react';
import LinkToPage from '../Components/LinkToPage';
import renderer from 'react-test-renderer';
import {BrowserRouter as Router} from "react-router-dom";

afterEach(() => {
    cleanup();
});

configure({
    adapter: new Adapter()
});

describe('LinkToPage', () => {
    const wrapper = shallow(<LinkToPage/>)
    it('renders correctly', () => {
        const tree = renderer.create(<Router><LinkToPage/></Router>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it('contains a link element', () => {
        expect(wrapper.find('Link')).toHaveLength(1);
    });
    it('contains an img element', () => {
        expect(wrapper.find('img')).toHaveLength(1);
    });
    it('changes colour', () => {
        const initialProps = {
            category: "things",
        }
        const container = shallow(<LinkToPage {...initialProps} />);
        expect(container.find('Link').props().to.state.color).toEqual("#D64539");
    });
});

