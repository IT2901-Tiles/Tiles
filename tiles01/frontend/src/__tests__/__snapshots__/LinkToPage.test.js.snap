// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkToPage renders correctly 1`] = `
<a
  href="/"
  onClick={[Function]}
>
  <img />
</a>
`;
